src/Surfnet/StepupSelfService/SelfServiceBundle/Controller/Registration/U2fController.php

```diff
@@ -0,0 +1,84 @@
+<?php
+
+/**
+ * Copyright 2014 SURFnet bv
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Surfnet\StepupSelfService\SelfServiceBundle\Controller\Registration;
+
+use Sensio\Bundle\FrameworkExtraBundle\Configuration\Template;
+use Surfnet\StepupSelfService\SelfServiceBundle\Command\ProveU2fDevicePossessionCommand;
+use Surfnet\StepupSelfService\SelfServiceBundle\Controller\Controller;
+use Surfnet\StepupSelfService\SelfServiceBundle\Service\U2fSecondFactorService;
+use Surfnet\StepupSelfService\SelfServiceBundle\Service\YubikeySecondFactorService;
+use Surfnet\StepupU2fBundle\Dto\RegisterResponse;
+use Symfony\Component\Form\FormError;
+use Symfony\Component\HttpFoundation\Request;
+
+class U2fController extends Controller
+{
+    /**
+     * @Template
+     */
+    public function provePossessionAction(Request $request)
```

Comment:
as discussed, might be wiser to split into separate GET and POST action handlers.